web_api/controllers/query_controller.py

```diff
@@ -38,81 +49,66 @@ def test_servers(body):  # noqa: E501
+    # Get whitelisted IPs from environmental variables...
+    whitelisted = False
     
-    #Validates token
-    if not check_token(token):
-        return ({"errorDesc": "Invalid token!"}, 400)
-
-    #Limits the rate at which the user may query the database 
-    if not check_time_limit(token):
-        return ({"errorDesc": "Too many queries in {0} seconds!".format(TIME_LIMIT)}, 400)
+    if os.environ.get("IP_WHITELIST"):
+        list1 = os.environ.get("IP_WHITELIST").split(" ")
+        if connexion.request.origin in list1:
+            whitelisted = True
+
+    # If IP is not whitelisted, verify the token and that rate limits are followed.
+    if not whitelisted:
+        #Checks if a token has been provided
+        if token == None or token == "":
+            return ({"errorDesc": "No token given!"}, 400)
+    
+        #Validates token
+        if not check_token(token):
+            return ({"errorDesc": "Invalid token!"}, 400)
+
+        #Limits the rate at which the user may query the database 
+        if not check_time_limit(token):
+            return ({"errorDesc": "Too many queries in {0} seconds!".format(TIME_LIMIT)}, 400)
+          
+        if captcha == None or captcha == "":
+            return  ({"errorDesc": "reCaptcha not checked"}, 400)
+    
+        valid_captcha = verify_captcha(captcha)
         
-            
-    # Now, we can start to run the checks. We define a list to which we append the results from each check.
-    checks_list = [checks.minimal_ns,
-                   checks.valid_hostname,
-                   checks.nameserver_reachability,
-                   checks.answer_authoritatively,
-                   checks.network_diversity,
-                   checks.consistency_glue_authoritative,
-                   checks.consistent_delegation_zone,
-                   checks.consistent_authoritative_nameservers,
-                   checks.truncref, checks.prohibited_networks,
-                   checks.dns_test_recursion, 
-                   checks.same_source_address]
-    results = []
+        if not valid_captcha[0]:
+            print(valid_captcha[1])
+            return  ({"errorDesc": "reCaptcha verification failed"}, 400)
 
-    # Run each check and append result to results.
-    for check in checks_list:
-        result = check.run(domain,name_servers)
-        # Check if the check returns a boolean or a more advanced dict consisting of a description too.
-        if isinstance(result, bool):
-            result = {"result": result, "description": str(check.__name__)}
-        results.append(result)
-
-    #Gives each check result a unique id and parses and combines them into the correct format
-    check_id = 0
-    list_of_check_results = []
-    for outcome in results:
-        details = outcome.get("details") if "details" in outcome else outcome.get("description")
-        list_of_check_results.append({"id":check_id, "result": outcome.get("result"), "key": details})
```

Comment:
This is the format for the json objects in the list that's added to the return with the "checks" key

Reply:
a more clear format was used upon agreement w/ Kasper since now on the front end the description will be pulled from the backend and will not be hardcoded.